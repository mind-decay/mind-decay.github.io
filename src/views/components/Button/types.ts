import { AnchorHTMLAttributes, ButtonHTMLAttributes, ReactNode } from 'react';

export type ButtonVariant = 'primary' | 'secondary';

type CommonType = {
  variant: ButtonVariant;
  children: ReactNode;
};

type AnchorTarget = '_target' | '_blank' | '_self' | '_parent' | '_top';

export type LinkType = CommonType &
  ButtonHTMLAttributes<HTMLButtonElement> & { onClick?: never; href?: never; target?: never; to: string };

export type ButtonType = CommonType &
  ButtonHTMLAttributes<HTMLButtonElement> & { onClick: () => void; href?: never; target?: never; to?: never };
export type AnchorType = CommonType &
  AnchorHTMLAttributes<HTMLAnchorElement> & {
    onClick?: never;
    href: string;
    target?: AnchorTarget;
    to?: never;
  };
