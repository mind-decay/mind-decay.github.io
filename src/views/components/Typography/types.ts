import { gradients } from 'styles/themes';

export type HeadingLevel = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
export type VariantType = Exclude<HeadingLevel, 'h5' | 'h6'>;

export type GradientName = keyof typeof gradients;

export type H2TitleProps = {
  $gradient?: GradientName;
};
