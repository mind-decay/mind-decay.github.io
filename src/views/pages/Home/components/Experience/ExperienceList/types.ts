export type ExperienceItem = {
  icons: string[];
  title: string;
  dateFrom: string;
  dateTo: string;
  Description: () => JSX.Element;
};
