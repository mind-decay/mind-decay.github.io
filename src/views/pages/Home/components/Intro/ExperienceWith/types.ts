import { FunctionComponent } from 'react';

export type SkillIcon = {
  Icon: FunctionComponent<React.SVGProps<SVGSVGElement>>;
  gradientId: string;
};
