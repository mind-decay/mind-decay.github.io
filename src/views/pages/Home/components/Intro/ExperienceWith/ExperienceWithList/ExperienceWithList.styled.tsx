import { ThemeType } from 'context/ThemeContext/types';
import styled from 'styled-components';
import { screenBreakpoints } from 'styles/utils';

export const StyledExperienceWithList = styled.ul<{ $itemsCount?: number }>`
  display: grid;
  grid-template-columns: repeat(${({ $itemsCount }) => ($itemsCount ? $itemsCount : 'auto-fit')}, max-content);
  column-gap: 6rem;
  list-style: none;

  @media screen and (max-width: ${screenBreakpoints.mobile}px) {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    column-gap: 3rem;
    row-gap: 1rem;
  }
`;

export const ExperienceWithItem = styled.li<{ $gradientId: string; $currentTheme: ThemeType }>`
  transform: scale(1);
  transition: 0.3s ease-in-out;

  svg {
    path {
      fill: ${({ $gradientId: gradientId, $currentTheme: currentTheme }) =>
        currentTheme === 'dark' ? `url(#${gradientId}_dark)` : `url(#${gradientId}_light)`};
    }
  }

  &:hover {
    transform: scale(1.1);
  }
`;
