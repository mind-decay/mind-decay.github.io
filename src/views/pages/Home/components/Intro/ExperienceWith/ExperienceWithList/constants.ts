import { SkillIcon } from '../types';

import JSLogo from 'assets/images/JSIcon.svg?react';
import NextJSIcon from 'assets/images/NextJSIcon.svg?react';
import HTMLIcon from 'assets/images/HtmlIcon.svg?react';
import CSSIcon from 'assets/images/CssIcon.svg?react';
import ReactIcon from 'assets/images/ReactIcon.svg?react';

export const skillsIcons: SkillIcon[] = [
  {
    gradientId: 'paint0_linear_14_58',
    Icon: JSLogo,
  },
  {
    gradientId: 'paint0_linear_14_59',
    Icon: NextJSIcon,
  },
  {
    gradientId: 'paint0_linear_14_60',
    Icon: HTMLIcon,
  },
  {
    gradientId: 'paint0_linear_14_61',
    Icon: CSSIcon,
  },
  {
    gradientId: 'paint0_linear_14_62',
    Icon: ReactIcon,
  },
];
