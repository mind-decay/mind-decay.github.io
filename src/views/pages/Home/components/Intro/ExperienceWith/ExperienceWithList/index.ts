export * from './ExperienceWithList';
export * from './ExperienceWithList.styled';
export * from './constants';
