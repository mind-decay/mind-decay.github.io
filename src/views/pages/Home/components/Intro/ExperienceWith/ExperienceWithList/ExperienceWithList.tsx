import { useTheme } from 'context/ThemeContext';

import { ExperienceWithItem, StyledExperienceWithList } from './ExperienceWithList.styled';

import { SkillIcon } from '../types';

type ExperienceWithListProps = {
  experiencesIcons: SkillIcon[];
};

export const ExperienceWithList = ({ experiencesIcons }: ExperienceWithListProps) => {
  const { theme } = useTheme();

  return (
    <StyledExperienceWithList $itemsCount={experiencesIcons.length}>
      {experiencesIcons.map(({ Icon, gradientId }) => {
        return (
          <ExperienceWithItem $gradientId={gradientId} $currentTheme={theme} key={gradientId}>
            <Icon />
          </ExperienceWithItem>
        );
      })}
    </StyledExperienceWithList>
  );
};
