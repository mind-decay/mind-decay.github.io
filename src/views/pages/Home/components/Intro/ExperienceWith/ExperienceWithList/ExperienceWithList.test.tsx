import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { ThemeProvider } from 'context/ThemeContext';

import { ExperienceWithList } from './ExperienceWithList';

import { SkillIcon } from '../types';

export const skillsIcons: SkillIcon[] = [
  {
    gradientId: '0',
    Icon: () => <svg></svg>,
  },
  {
    gradientId: '1',
    Icon: () => <svg></svg>,
  },
  {
    gradientId: '2',
    Icon: () => <svg></svg>,
  },
  {
    gradientId: '3',
    Icon: () => <svg></svg>,
  },
  {
    gradientId: '4',
    Icon: () => <svg></svg>,
  },
];

test('renders ExperienceWithList component', () => {
  const { container } = render(
    <ThemeProvider>
      <ExperienceWithList experiencesIcons={skillsIcons} />
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});
