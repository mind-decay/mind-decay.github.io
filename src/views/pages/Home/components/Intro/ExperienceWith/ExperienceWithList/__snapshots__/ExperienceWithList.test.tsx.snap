// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders ExperienceWithList component 1`] = `
<div>
  <ul
    class="sc-beySPh iPxRCy"
  >
    <li
      class="sc-guDLey dKvQr"
    >
      <svg />
    </li>
    <li
      class="sc-guDLey gfWWLe"
    >
      <svg />
    </li>
    <li
      class="sc-guDLey jJoStp"
    >
      <svg />
    </li>
    <li
      class="sc-guDLey lalvQY"
    >
      <svg />
    </li>
    <li
      class="sc-guDLey fzuWWT"
    >
      <svg />
    </li>
  </ul>
</div>
`;
